### ISSUE TITLE
Stream Category laws

### ISSUE THREAD

**Issue by dakom** [SodiumFRP/purescript-sodium#17]
Should follow just like the Cell tests, but we need to define an Eq that works through Aff, since the only way of getting a value out of Stream is via `listen`

**Comment by dakom** [comment]
Asked question on quickcheck-laws here: https://github.com/garyb/purescript-quickcheck-laws/issues/44

**Comment by clinuxrulz** [comment]
Sodium is synchronous. At the end of a transaction all listener callbacks that are required to be invoked are invoked.

**Comment by dakom** [comment]
Right on - Ref's are used in various tests, e.g. here: https://github.com/SodiumFRP/purescript-sodium/blob/master/test/Stream.test.purs#L69

Still not sure how to wrap around this in an `Eq` instance though... will need to play with it and see at some point :)

**Comment by clinuxrulz** [comment]
Yep. No nice way to do this. You'll have to play dirty with unsafePerformEffect or design something similar to quickcheck that is more suitable for sodium.

Maybe turning streams into cells holding a list of tuples of the current transaction count a long with the value at that moment could work.
